scala-package/core/src/main/scala/ml/dmlc/mxnet/IO.scala

```diff
@@ -95,8 +95,8 @@ object IO {
   private[mxnet] def initData(data: IndexedSeq[NDArray],
                               allowEmpty: Boolean,
                               defaultName: String): IndexedSeq[(String, NDArray)] = {
-    require(data != null || allowEmpty)
-    if (data == null) {
```

Comment:
then do the assertion, `require(data != null)`, same for those in `NDArrayIter`.